crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -414,9 +415,21 @@ where
         &self,
         number: BlockNumberOrTag,
         full: bool,
+        include_header: Option<bool>,
     ) -> RpcResult<Option<RichBlock>> {
         trace!(target: "rpc::eth", ?number, ?full, "Serving eth_getBlockByNumber");
-        Ok(EthBlocks::rpc_block(self, number.into(), full).await?)
+        let mut block = EthBlocks::rpc_block(self, number.into(), full).await?;
+
+        if let Some(ref mut rich_block) = block {
+            if include_header.is_some_and(|v| v) {
+                // Add the header JSON to extra_info
+                let header_json = serde_json::to_value(&rich_block.header)
+                    .map_err(|e| internal_rpc_err(format!("Failed to serialize header: {}", e)))?;
+                rich_block.extra_info.insert("headerJson".to_string(), header_json);
```

Comment:
Probably dont need the JSON ending as that is implied in the response
```suggestion
                rich_block.extra_info.insert("extraDataHeader".to_string(), header_json);
```